### DISCUSSION TITLE
Nested Form Question

### DISCUSSION THREAD

**Discussion by rachelrj** [react-hook-form/react-hook-form#8602] [General]
I am building a form that has a nested form that conditionally appears.
I am wondering whether to use `useForm` or `useFormContext` for the nested form.

If using `useFormContext` is there a way to send the default values for the nested fields back up to the parent form?
Or do the default values need to be on the inputs themselves.

I've tried using both `useForm` and `useFormContext` and when the conditional inputs unmount or remount the validation becomes incorrect.
I am using `useController` for each input and setting `shouldUnregister` to be true for the nested fields. (So, I am not explicitly registering each field)

**Comment by shamseerahammedm** [comment]
what do u mean validation becomes incorrect? `useFormContext` is just for taking the methods of `useForm` to nested components of your form as far as I know, which means u have to `useForm` irrespective of weather its nested or normal form

**Reply by rachelrj** [reply]
Hello @shamseerahammedm 
Yes, I am already using `useForm`. My question is whether I can have a nested `useForm`.
The validation becoming incorrect has to do with the `watch` method, somewhat like @zemariagp explains in his comment.
But, I'll follow up with an explanation that is more detailed about the incorrectness.

**Comment by zemariagp** [comment]
i think i know what @rachelrj means, because i might be experiencing the same.
i have a group of checkboxes being registered, select a few, then i unmount the component, mount it again, the form state is still there but when i select/deselect some checkboxes, the the watch method for this field is adding too many values.

**Reply by rachelrj** [reply]
Yes, the issue seemed to have to do with the `watch` methods, but for other components within the form.
